assets/javascripts/discourse-assign/controllers/assign-user.js

```diff
@@ -1,12 +1,13 @@
 import Controller, { inject as controller } from "@ember/controller";
+import ModalFunctionality from "discourse/mixins/modal-functionality";
 import { action } from "@ember/object";
 import { not, or } from "@ember/object/computed";
 import { inject as service } from "@ember/service";
 import { isEmpty } from "@ember/utils";
 import { ajax } from "discourse/lib/ajax";
 import { popupAjaxError } from "discourse/lib/ajax-error";
 
-export default Controller.extend({
+export default Controller.extend(ModalFunctionality, {
```

Comment:
What does ModalFunctionality add here?

Reply:
Mostly added it for consistency with core, here is what it adds, pretty minimal https://github.com/discourse/discourse/blob/f4b45df83fe8b1b63d1725005674a2d1b1a2f8ca/app/assets/javascripts/discourse/app/mixins/modal-functionality.js